packages/ui/src/memberships/modals/BuyMembershipModal/BuyMembershipFormModal.tsx

```diff
@@ -293,54 +327,46 @@ export const BuyMembershipForm = ({
                         If your validator account is not in your signer wallet, paste the account address to the field
                         below:
                       </TextMedium>
-                      <RowInline>
-                        <InputComponent id="select-validatorAccount" inputSize="l">
-                          <SelectAccount
-                            id="select-validatorAccount"
-                            name="validatorAccountCandidate"
-                            filter={(account) => !!validatorAddresses?.includes(account.address)}
-                          />
-                        </InputComponent>
-                        <ButtonPrimary
-                          square
-                          size="large"
-                          onClick={addValidatorAccount}
-                          disabled={!isValidValidatorAccount}
-                          className="add-button"
-                        >
-                          <PlusIcon />
+                      {validatorAccountCandidates?.map((account, index) => (
+                        <Row>
+                          <RowInline>
+                            <InputComponent id="select-validatorAccount" inputSize="l">
+                              <SelectAccount
+                                id="select-validatorAccount"
+                                name={`validatorAccountCandidates[${index}]`}
+                                filter={(account) => validatorAccountSelectorFilter(index, account)}
+                              />
```

Comment:
But for all this to work this should become a controlled input:
```suggestion
                              <BaseSelectAccount
                                value={validatorAccounts[index]}
                                onChange={(account)) => addValidatorAccount(index, account)}
                                filter={(account) => validatorAccountSelectorFilter(index, account)}
                              />
```

Also please don't use a static id in loops, it makes invalid html and react complains about that.